### PR TITLE
autodiff recipe

### PR DESCRIPTION
**Pull request by allanleal**

### CODE REVIEW

File: recipes/autodiff/LICENSE [hunk 1]
```diff
@@ -0,0 +1,21 @@
+MIT License
```
Comment (chrisburr) [inline recipes/autodiff/LICENSE]: You can delete this file as it's already included in the tarball.

File: recipes/autodiff/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+  build:
+    - {{ compiler('cxx') }}
+    - cmake
+    - ninja
+
+  host:
+    - eigen
+
+  run:
+    - eigen
```
Comment (chrisburr) [inline recipes/autodiff/meta.yaml]: Should this be [pinned](https://docs.conda.io/projects/conda-build/en/latest/source/resources/variants.html?highlight=pin_compatible#pinning-at-the-recipe-level) to a range of `eigen` versions using something like:

```suggestion
    - {{ pin_compatible('eigen', max_pin='x.x') }}
```

This means that if this was build with `eigen==3.3.7` the runtime version must be meet the requirements `>=3.3.0` and `<3.4`. I don't know what the appropriate pinning is but it's probably one of: `x`, `x.x` or `x.x.x`.

Comment (allanleal) [reply recipes/autodiff/meta.yaml]: Honestly, I don't know. I checked all recipes in `conda-recipes` and it seems the usage is:
~~~
  - eigen
  - eigen3
  - eigen 3.*
~~~
Maybe use `eigen 3.*`? Or just stick with `eigen` if this means the same?

File: recipes/autodiff/LICENSE (hunk 1 above)
Comment (allanleal) [reply recipes/autodiff/LICENSE]: @chrisburr Thanks for reviewing this recipe! I'll delete this file.

**Comment by allanleal** [comment]
I see there are some recipes along with mine. How to I get rid of them?

File: recipes/autodiff/meta.yaml (hunk 2 above)
Comment (chrisburr) [reply recipes/autodiff/meta.yaml]: A package called `eigen3` [doesn't exist](https://anaconda.org/conda-forge/eigen3). I've looked elsewhere and it's unusual for eigen to be a runtime dependency but I guess it is actually needed by `autodiff`?

If it is and if you expect any version of eigen 3 to work I would use:
```yaml
  host:
    - eigen
  run:
    - {{ pin_compatible('eigen', max_pin='x') }}
```

**Comment by chrisburr** [comment]
> I see there are some recipes along with mine. How to I get rid of them?

I'm not entirely sure what went wrong with the merge commits but I think I've fixed it. When you make a commit to fix the `eigen3` error everything should be working again.

**Comment by nicoddemus** [comment]
cc @igortg

**Comment by allanleal** [comment]
Hi @chrisburr , all done  - replaced `eigen3` to `eigen` and removed `eigen` from runtime dependency. 

All is passing but this travis build (Xcode: xcode9.4) that is always pending for hours!

You asked before about support to osx and windows in addition to linux. Yes, this library has been fixed to be compiled in all these systems. I forgot to remove `skip` for `win` and `osx`. Hopefully when you see this message the new build will be finished already.

**Comment by chrisburr** [comment]
Yeah the Travis backlog is really bad, hopefully Azure can take over at some point and lessen the load.

You can ignore the `win_64` build on Azure as that is known to have problems but it looks like the Appveyor Windows build is also failing.

**Comment by allanleal** [comment]
We need `Visual Studio 15 2017` and not `Visual Studio 14 2015` to successfully compile `autodiff`, because of C++17 requirements. This is what I was afraid and contacted gitter last week. 

**AppVeyor**
~~~
%SRC_DIR%>SET "CMAKE_GENERATOR=Visual Studio 14 2015 Win64"  
%SRC_DIR%>CALL "C:\Program Files (x86)\Microsoft Visual Studio 14.0\Common7\IDE\..\..\VC\vcvarsall.bat" amd64  
-- The CXX compiler identification is MSVC 19.0.24241.7
-- Check for working CXX compiler: C:/Program Files (x86)/Microsoft Visual Studio 14.0/VC/bin/amd64/cl.exe
-- Check for working CXX compiler: C:/Program Files (x86)/Microsoft Visual Studio 14.0/VC/bin/amd64/cl.exe -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
~~~

File: recipes/autodiff/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+package:
+  name: autodiff
+  version: {{ version }}
+
+source:
+  url: https://github.com/autodiff/autodiff/archive/v{{ version }}.tar.gz
+  sha256: 3f8246b8bc38bdfa41860c9a5e909b55793d2455edc525bdaa20b8060667b706
+
+build:
+  number: 0
```
Comment (chrisburr) [inline recipes/autodiff/meta.yaml]: I think this should do it:
```suggestion
  number: 0
  skip: True  # [win and vc<15]
```

Comment (allanleal) [reply recipes/autodiff/meta.yaml]: I pressed `Commit suggestion` but it did not change the file.

File: recipes/autodiff/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+  skip: True  # [win and vc<15]
+  
+requirements:
+  build:
+    - {{ compiler('cxx') }}
```
Comment (chrisburr) [inline recipes/autodiff/meta.yaml]: `clang++` requires `clang` on macOS:
```suggestion
    - {{ compiler('c') }}
    - {{ compiler('cxx') }}
```

**Comment by allanleal** [comment]
@chrisburr  The build passed for osx here:
https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=15433

but when compiling the test example, it failed at the cmake configure stage:
~~~
-- The C compiler identification is Clang 4.0.1
-- The CXX compiler identification is Clang 4.0.1
-- Check for working C compiler: $PREFIX/bin/x86_64-apple-darwin13.4.0-clang
-- Check for working C compiler: $PREFIX/bin/x86_64-apple-darwin13.4.0-clang -- broken
~~~

File: recipes/autodiff/meta.yaml (hunk 4 above)
Comment (allanleal) [reply recipes/autodiff/meta.yaml]: Great - thanks. Let's see now if it passes for osx.

File: recipes/autodiff/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+
+  run:
+    - eigen
+    
+test:
+  files:
+    - test/
+
+  requires:
+    - {{ compiler('cxx') }}
```
Comment (chrisburr) [inline recipes/autodiff/meta.yaml]: And here:
```suggestion
    - {{ compiler('c') }}
    - {{ compiler('cxx') }}
```

File: recipes/autodiff/meta.yaml (hunk 2 above)
Comment (allanleal) [reply recipes/autodiff/meta.yaml]: I missed this. I'll add:
~~~yaml
  run:
    - {{ pin_compatible('eigen', max_pin='x') }}
~~~

File: recipes/autodiff/meta.yaml [hunk 6]
```diff
@@ -39,7 +39,6 @@ about:
   home: https://autodiff.github.io
   license: MIT
   license_family: MIT
-  license_file: LICENSE
```
Comment (chrisburr) [inline recipes/autodiff/meta.yaml]: Why did you remove this? Not including it in the package is a violation of the licence which states:

> The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.

Comment (allanleal) [reply recipes/autodiff/meta.yaml]: Hmm I thought that was only needed if the `LICENSE` file is inside the `recipe` dir.

Comment (chrisburr) [reply recipes/autodiff/meta.yaml]: Yeah if you put `[ci skip]` in the commit message it will skip it

Comment (allanleal) [reply recipes/autodiff/meta.yaml]: I'll do this then because I guess the issue with osx_64 is now solved (well, skipped - the `test_run.sh` exits if `CLANGXX` is defined!).

Comment (chrisburr) [reply recipes/autodiff/meta.yaml]: Oh and it's also picked up from the work directory, i.e. the directory used for the build step containing the source. Conda-build might pick it up by default in this case, but explicitly listing its filename will make the CI fail if it is missing rather than risking it being missed.

Comment (allanleal) [reply recipes/autodiff/meta.yaml]: OK, all has passed (except the travis build that is still pending).

**Comment by chrisburr** [comment]
Thanks, LGTM 👍

**Comment by allanleal** [comment]
Thanks @chrisburr :+1:
